chore: #188 error messaging

NoWhere-2-Go opened the pull request:
## Overview
Added a error message when a search yielded no results or whenever there is an issue with the API

## What Changed
Previously if there were issues with the api or if a search yielded no results the program displayed no error messages to the user. 

## To-Do:

I need to make the message display more detailed messages depending on if there is an issue with the API or something else

NoWhere-2-Go committed:
Merge remote-tracking branch 'upstream/master' into develop

NoWhere-2-Go committed:
Update search.ts

NoWhere-2-Go committed:
Added error UI 

Need to add different messages depending upon error. But for now the program will display this error alert to user if the API has an issue or if the course doesn't exist

ZzRanger commented:
Addresses #188 . In the future, I'd suggest linking the relevant issue/s to PRs so that you include the context behind the PR.

ZzRanger commented:
Okay at a glance you have the right idea! You're creating the Alert and calling it whenever an error state is reached. I'll go over this with you tomorrow during the meeting to explain how to bubble up the error from the useSearch hook itself.

NoWhere-2-Go committed:
feat(errrormessage.tsx): Adds error messaging UI when useSearch fails

Adds error messaging UI when useSearch fails, when and if it happens an alert is
generate onto the web page

chore: #188

NoWhere-2-Go committed:
chore: #188

NoWhere-2-Go committed:
Merge branch 'develop' of https://github.com/UTDNebula/planner into develop

NoWhere-2-Go committed:
Update CourseSelector.tsx

NoWhere-2-Go committed:
Update CourseSelector.tsx

NoWhere-2-Go committed:
Merge branch 'develop' of https://github.com/NoWhere-2-Go/planner into develop
